shell/platform/darwin/ios/framework/Source/FlutterRestorationPluginTest.mm

```diff
@@ -80,7 +80,7 @@ - (void)testRestorationEnabledWaitsForData {
   [restorationPlugin setRestorationData:data];
   XCTAssertEqual([capturedResult count], 2u);
   XCTAssertEqual([capturedResult objectForKey:@"enabled"], @YES);
-  XCTAssertEqual([[capturedResult objectForKey:@"data"] data], data);
+  XCTAssertEqualObjects([[capturedResult objectForKey:@"data"] data], data);
```

Comment:
@goderbauer Could you take a look at the changes I made in this test file to see if they are ok? Did you remember if `XCTAssertEqual` was intentionally used to compare the address of the 2 data objects, or we just need the content of the `data` objects to be the same.

Original PR: https://github.com/flutter/engine/pull/23495

Reply:
We just need the content of the data objects to be the same.